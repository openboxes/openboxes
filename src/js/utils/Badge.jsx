import React from 'react';

import PropTypes from 'prop-types';

import './utils.scss';

const Badge = ({ label, variant }) => (
  <span className={`badge ${variant}`}>{label}</span>
);

Badge.propTypes = {
  label: PropTypes.string.isRequired,
  variant: PropTypes.string.isRequired,
};

export default Badge;
