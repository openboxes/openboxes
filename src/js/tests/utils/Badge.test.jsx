import React from 'react';

import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';

import Badge from 'utils/Badge';

import '@testing-library/jest-dom';

jest.mock('wrappers/CustomTooltip', () => ({ children }) => <>{children}</>);

let badgePrimary;
let badgeWithTooltip;
let badgeEmptyVariant;

describe('Badge component', () => {
  beforeEach(() => {
    badgePrimary = <Badge label="Primary Badge" variant="primary" />;
    badgeWithTooltip = <Badge label="Tooltip Badge" variant="warning" tooltip />;
    badgeEmptyVariant = <Badge label="Empty Variant" variant="" />;
  });

  it('should match snapshot', () => {
    const badge = renderer.create(badgeWithTooltip).toJSON();
    expect(badge).toMatchSnapshot();
  });

  it('renders the label text', () => {
    render(badgePrimary);
    expect(screen.getByText('Primary Badge')).toBeInTheDocument();
  });

  it('applies the correct variant class', () => {
    render(badgePrimary);
    const badgeElement = screen.getByTestId('badge');
    expect(badgeElement).toHaveClass('badge');
    expect(badgeElement).toHaveClass('primary');
  });

  it('renders nothing when label prop is undefined', () => {
    const consoleSpy = jest.spyOn(console, 'error').mockImplementation(() => {});
    render(<Badge variant="primary" />);
    expect(screen.queryByTestId('badge')).not.toBeInTheDocument();
    consoleSpy.mockRestore();
  });

  it('applies only base class when variant is empty string', () => {
    render(badgeEmptyVariant);
    const badgeElement = screen.getByTestId('badge');
    expect(badgeElement).toHaveClass('badge');
    expect(badgeElement).not.toHaveClass('primary', 'success', 'info', 'warning', 'danger');
  });
});
