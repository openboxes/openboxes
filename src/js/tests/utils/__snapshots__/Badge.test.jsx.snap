// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class=""
  >
    <div
      aria-describedby="tippy-tooltip-1"
      class="w-100"
      data-original-title="undefined"
      data-tooltipped=""
      style="display: inline;"
    >
      <div
        class="flex items-center"
      >
        <span
          class="badge warning"
          data-testid="badge"
        >
          Snapshot Test
        </span>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Badge component should match snapshot 1`] = `
<span
  className="badge warning"
  data-testid="badge"
>
  Tooltip Badge
</span>
`;
